include/crocoddyl/core/optctrl/shooting.hxx

```diff
@@ -520,11 +521,24 @@ std::size_t ShootingProblemTpl<Scalar>::get_nu_max() const {
 template <typename Scalar>
 std::size_t ShootingProblemTpl<Scalar>::get_nthreads() const {
 #ifndef CROCODDYL_WITH_MULTITHREADING
-  std::cerr << "Warning: the number of threads won't affect the computational performance as it is not enable the "
-               "multithreading support."
+  std::cerr << "Warning: the number of threads won't affect the computational performance as multithreading "
+               "support is not enabled."
             << std::endl;
 #endif
   return nthreads_;
 }
 
+template <typename Scalar>
+std::ostream& operator<<(std::ostream& os, const ShootingProblemTpl<Scalar>& problem) {
+  os << "ShootingProblem (T=" << problem.get_T() << ", nx=" << problem.get_nx() << ", ndx=" << problem.get_ndx()
+     << ", nu_max=" << problem.get_nu_max() << ") " << std::endl;
+  os << "  Models:" << std::endl;
+  auto runningModels = problem.get_runningModels();
```

Comment:
Please do not use 'auto' as it is not supported by c++98 (we did not decide yet to move to c++11)

Reply:
Thanks for the feedback - I'll make the amendments throughout. With 16.04 support ending next month it may be worth discussing moving to a newer C++ standard unless there is a requirement to support C++98?

Reply:
Yes, indeed. Thank you for opening the issue ;)

Reply:
This conversation continues in #934